R/dataset.R

```diff
@@ -88,6 +88,56 @@ setMethod("notes<-", "CrunchDataset", function (x, value) {
     invisible(setEntitySlot(x, "notes", value))
 })
 
+#' Get and set the market size for Crunch datasets
+#'
+#' Crunch Datasets allow you to set a target population size in order to extrapolate
+#' population estimates from survey percentages. This function lets you work with
+#' the population size and magnitude.
+#' @param x a Crunch Dataset
+#' @param size the target population size
+#' @param magnitude the order of magnitude with which to display the population
+#' size. Must be either `3`, `6`, or `9` for thousands, millions, and billions respectively.
+#' @return `popSize` and `popMagnitude` return the population size or
+#' magnitude. `setPopultion` returns the modified dataset.
+#' @name population
+#' @aliases popSize popMagnitude setPopulation
+NULL
+
+#' @rdname population
+#' @export
+setMethod("popSize", "CrunchDataset", function (x) {
+    return(settings(x)$population$size)
+})
+
+#' @rdname population
+#' @export
+setMethod("popMagnitude", "CrunchDataset", function (x) {
+    return(settings(x)$population$magnitude)
+})
```

Comment:
Would it make sense to expose `popSize<-` and `popMagnitude<-` methods to let users update size/magnitude? I'm not 100% set that we should add it, just a pattern I know we support elsewhere (and in classic R fashion: why have one way to do something when you can have multiple?). There's a complication for the initial setting that you need both at the same time, but after they're set, you should be able to `PATCH` either independently. 

Something like:
```
setMethod("popSize<-", "CrunchDataset", function (x, value) {
    if (is.null(settings(x)$population$size)) {
        halt("Please set with `setPopulation(ds, ...)`)
    }
    settings(x)$population$size <- value
    return(invisible(x))
}
```

Reply:
My intuition is that if we're going to do an assignment method it should work on all the datasets and having the user switch back and forth between two different ways of setting something based on the dataset state is a bit confusing.  It seems like this is mostly going to come up with people setting populations on a dataset for the first time so in that case we could just default to a magnitude of thousands. What about something like this?:
```
setMethod("popSize<-", "CrunchDataset", function (x, value) {
    if (is.null(settings(x)$population$magnitude)) {
        warning("magnitude not set, defaulting to thousands")
        setPopulation(x, value, 3)
    }
    settings(x)$population$size <- value
    return(invisible(x))
}
```
This doesn't work for setting magnitude because we don't want to default to a population, but that's probably not going to come up as much.

Reply:
That works for me. I was mostly wanting to give R users an easy way to adjust both the size and magnitude without adjusting the other. So for `popMagnitude<-` we would just outright error if it's not set, but users could still change magnitudes easily after the fact.